neuralpredictors/layers/encoders/poisson.py

```diff
@@ -0,0 +1,34 @@
+from torch import nn
+
+
+class PoissonEncoder(nn.Module):
```

Comment:
Now that I think of it - I think it could be nice to have a more abstract base class which is just called Encoder, without the fancy stuff. So just core/readout/elu_offset. Maybe sitting in base.py. Then it's a consistent structure.
The PoissonEncoder can then inherit from it.

Reply:
Mh, but what would we outsource to that base class? Everything that the PoissonEncoder has, is already very specific

Reply:
At least the init can be inherited. So that the base class takes care of initializing core/readout/offset. And with the `super().__init__`, the base class always has that job. I think the overall advantage is that it is very easy to understand what the base class is doing, if the forward is also so simple.
Something like:
```

x = self.core(inputs)
x = self.readout(x, data_key=data_key, shift=shift, **kwargs)
return nn.functional.elu(x + self.offset) + 1
```

Reply:
then the poisson class would overwrite the forward, but it's clean and understandable at first glance.

Reply:
But that code you sent there is also specific to the PoissonEncoder, it has the `elu + 1` which is designed to represent a firing rate, the parameter of the poisson distribution. So the `forward` can not be outsourced, it always depends on the distribution. 
And the `__init__` would not gain anything from outsourcing to a base class codewise because you would need to split up the arguments and pass some of them to `super().__init__()` and others to init itself... That's just as much code... I guess one culd make an argument that it is slightly easier to understand. But then again it's really not much and one needs to look up a completely new file just to see 2 lines of code where the core and readout arguments are turned into a class attribute... 
Not sure, any opinions?

Reply:
Yea you are right, the `+ 1` is a poisson thing. I still like the base class. I'm having the same `__init__` issue with the cores. With one base class having the init, and the children have very similar `__inits__`s.

I'm solving it by doing:
in the base class:
`def __init__(arg1, arg2, arg3=1):`...

and then in the children:
```
def __init__(*args, arg4=100, arg5=200, **kwargs):
   super().__init__(*args, **kwargs)
   self.arg4=arg4
   self.arg5=arg5
   ```

Reply:
Then the base class still has its role in the init, and the children will only initialize what they add on top.

Reply:
So in the meeting we agreed on not having a base class for the moment. I'll close this conversation for now